Added package Evodiff and dependencies

ashim-mahara opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

ashim-mahara committed:
added py-evodiff and dependencies

ashim-mahara committed:
deleted the FIXME

ashim-mahara committed:
fixed style issues

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    depends_on("py-hatch", type="build")
+    depends_on("py-hatch-cython", type="build")
+
+    depends_on("py-numpy", type=("build", "run"))
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    depends_on("py-hatch-cython", type="build")
+
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-msgpack", type=("build", "run"))
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-msgpack", type=("build", "run"))
+    depends_on("py-biotraj", type=("build", "run"))
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-msgpack", type=("build", "run"))
+    depends_on("py-biotraj", type=("build", "run"))
+    depends_on("py-requests", type=("build", "run"))
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    depends_on("py-msgpack", type=("build", "run"))
+    depends_on("py-biotraj", type=("build", "run"))
+    depends_on("py-requests", type=("build", "run"))
+    depends_on("py-networkx", type=("build", "run"))
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+
+    version("1.0.1", sha256="7012158431fd488c26d78d33032550eea1d7af7afd01b48549a7fd239f63dab5")
+
+    depends_on("py-python-dotenv", type="build")
```
Where did this come from?

ashim-mahara replied:
removed

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    version("1.0.1", sha256="7012158431fd488c26d78d33032550eea1d7af7afd01b48549a7fd239f63dab5")
+
+    depends_on("py-python-dotenv", type="build")
+    depends_on("py-hatch-vcs", type="build")
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+
+    depends_on("py-python-dotenv", type="build")
+    depends_on("py-hatch-vcs", type="build")
+    depends_on("py-hatchling", type="build")
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    depends_on("py-python-dotenv", type="build")
+    depends_on("py-hatch-vcs", type="build")
+    depends_on("py-hatchling", type="build")
+    depends_on("py-hatch", type="build")
```
Missing version

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    depends_on("py-hatch-vcs", type="build")
+    depends_on("py-hatchling", type="build")
+    depends_on("py-hatch", type="build")
+    depends_on("py-hatch-cython", type="build")
```
Missing version

ashim-mahara replied:
no versions in package descriptions.

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    library."""
+
+    homepage = "https://www.biotite-python.org/latest/"
+    pypi = "biotite/biotite-1.0.1.tar.gz"
```
~It looks like you're pulling from https://github.com/biotite-dev/biotite/releases/tag/v1.0.1 instead of PyPI, so that needs to be reflected here, `git = "git@github.com:biotite-dev/biotite.git"`~

My mistake, when I looked at PyPI, I was looking at "biolite". You can ignore this.

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,33 @@
+    license("BSD-3-Clause")
+
+    version("1.0.1", sha256="7012158431fd488c26d78d33032550eea1d7af7afd01b48549a7fd239f63dab5")
+
```
Missing `depends_on("python@3.10:", type=("build", "run"))`

meyersbs reviewed (changes requested):
I only reviewed one package. It looks like most of the packages you have are missing versions for dependencies. Please go through and check them all.

ashim-mahara committed:
added versions for biotite dependencies; added hash to py-hatch-vcs

ashim-mahara committed:
added python version for py-hatch-cython

ashim-mahara committed:
updated biotraj dependencies

ashim-mahara committed:
- added versions for the packages and dependencies
- added more dependencies for py-hatch
- added rust versions
- added py-uv as a new package

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,34 @@
+
+    depends_on("py-numpy@1.25:", type=("build", "run"))
+    depends_on("py-msgpack@0.5.6:", type=("build", "run"))
+    depends_on("py-biotraj@1.0:2.0", type=("build", "run"))
```
Should be `@1:1` (this is any version from 1.0 to 1.999999...., but not 2.0)

ashim-mahara replied:
okay thanks , didn't know that.

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,34 @@
+
+    depends_on("python@3.10:", type=("build", "run"))
+
+    depends_on("py-hatch-vcs@0.3.0:", type="build")
```
From pyproject.toml: `"hatch-vcs == 0.4",`

meyersbs commented on var/spack/repos/builtin/packages/py-biotite/package.py:
```diff
@@ -0,0 +1,34 @@
+from spack.package import *
+
+
+class PyBiotite(PythonPackage):
```
Missing dependency for cython >= 3.0

meyersbs commented on var/spack/repos/builtin/packages/py-biotraj/package.py:
```diff
@@ -0,0 +1,26 @@
+
+    license("LGPL-2.1")
+
+    version("1.2.1", sha256="4c93641dfa138d067be8d3aeb8f07be6342ec76d450ad3ab1801a92bfdbfbf85")
```
Missing `requires-python = ">=3.10"` and `"setuptools_scm>=8",`

meyersbs commented on var/spack/repos/builtin/packages/py-biotraj/package.py:
```diff
@@ -0,0 +1,26 @@
+
+    version("1.2.1", sha256="4c93641dfa138d067be8d3aeb8f07be6342ec76d450ad3ab1801a92bfdbfbf85")
+    version("1.2.0", sha256="d2226348546ef384d23f608cca8f7110d1e6fa8927dcbd74dbf00ee748929cd1")
+    version("1.1.0", sha256="462f67011a2953f6b4e7c835a658c7c89e68aa89f00dfa64f98679ddc8d64ea3")
```
Version 1.2.0 and 1.1.0 have different dependencies, either add them and use `when="@version"` or remove these older definitions from the package recipe.

ashim-mahara replied:
removed the 1.1.0 version

meyersbs commented on var/spack/repos/builtin/packages/py-evodiff/package.py:
```diff
@@ -0,0 +1,33 @@
+
+    license("MIT", checked_by="ashim-mahara")
+
+    version("1.1.0", sha256="c1f2d7bd0e46ad244f1c55066caefc5ad9b1bcf4e836be1832311b8cd74e923f")
```
Missing `py-mlflow` dependency

ashim-mahara replied:
https://github.com/microsoft/evodiff/pull/44

ashim-mahara replied:
they removed the dependency, i couldn't find it in the main branch

ashim-mahara replied:
@meyersbs

meyersbs replied:
Don't look at the branch, look at the .tar.gz from the release: https://files.pythonhosted.org/packages/52/9a/9d4ee39c6d6fbc179b36c36db80b4bf37c8dc91f3fad38a3736472c1ce1f/evodiff-1.1.0.tar.gz

```
    install_requires=[
        "pandas",
        "lmdb",
        "numpy",
        "sequence-models",
        "mlflow",
        "scikit-learn",
        "blosum",
        "seaborn",
        "matplotlib",
        "fair-esm",
        "tqdm",
        "biotite",
        "requests",
        "mdanalysis",
        "pdb-tools"
    ],
```

ashim-mahara replied:
done

meyersbs commented on var/spack/repos/builtin/packages/py-evodiff/package.py:
```diff
@@ -0,0 +1,33 @@
+    license("MIT", checked_by="ashim-mahara")
+
+    version("1.1.0", sha256="c1f2d7bd0e46ad244f1c55066caefc5ad9b1bcf4e836be1832311b8cd74e923f")
+
```
From pyproject.toml, missing: `requires = ["setuptools>=61.0"]`

meyersbs commented on var/spack/repos/builtin/packages/py-fair-esm/package.py:
```diff
@@ -0,0 +1,28 @@
+
+    license("MIT")
+
+    version("2.0.0", sha256="4ed34d4598ec75ed6550a4e581d023bf8d4a8375317ecba6269bb68135f80c85")
```
Looking at the setup.py, `esmfold` should be a variant with the dependencies listed under that

```
variant("esmfold", default=True, description="Enable dependencies for OpenFold")
depends_on("py-biopython", when="+esmfold", type=("build", "run"))
...
# other deps here
```

ashim-mahara replied:
done

meyersbs commented on var/spack/repos/builtin/packages/py-fair-esm/package.py:
```diff
@@ -0,0 +1,28 @@
+
+    depends_on("py-setuptools@59.5.0:", type=("build"))
+
+    depends_on("py-biopython@1.79:", type=("build", "run"))
```
Where did this version come from?

ashim-mahara replied:
https://github.com/facebookresearch/esm/blob/main/environment.yml

meyersbs commented on var/spack/repos/builtin/packages/py-fair-esm/package.py:
```diff
@@ -0,0 +1,28 @@
+
+    depends_on("py-biopython@1.79:", type=("build", "run"))
+    depends_on("py-deepspeed@0.5.9:", type=("build", "run"))
+    depends_on("py-dm-tree@0.1.6:", type=("build", "run"))
```
Where did this version come from?

ashim-mahara replied:
https://github.com/facebookresearch/esm/blob/main/environment.yml

meyersbs commented on var/spack/repos/builtin/packages/py-fair-esm/package.py:
```diff
@@ -0,0 +1,28 @@
+    depends_on("py-biopython@1.79:", type=("build", "run"))
+    depends_on("py-deepspeed@0.5.9:", type=("build", "run"))
+    depends_on("py-dm-tree@0.1.6:", type=("build", "run"))
+    depends_on("py-pytorch-lightning@1.5.10:", type=("build", "run"))
```
Where did this version come from?

meyersbs commented on var/spack/repos/builtin/packages/py-fair-esm/package.py:
```diff
@@ -0,0 +1,28 @@
+    depends_on("py-dm-tree@0.1.6:", type=("build", "run"))
+    depends_on("py-pytorch-lightning@1.5.10:", type=("build", "run"))
+    depends_on("py-omegaconf", type=("build", "run"))
+    depends_on("py-ml-collections@0.1.0:", type=("build", "run"))
```
Where did this version come from?

ashim-mahara replied:
https://github.com/facebookresearch/esm/blob/main/environment.yml

meyersbs commented on var/spack/repos/builtin/packages/py-fair-esm/package.py:
```diff
@@ -0,0 +1,28 @@
+    depends_on("py-omegaconf", type=("build", "run"))
+    depends_on("py-ml-collections@0.1.0:", type=("build", "run"))
+    depends_on("py-einops", type=("build", "run"))
+    depends_on("py-scipy@1.7.1:", type=("build", "run"))
```
Where did this version come from?

ashim-mahara replied:
https://github.com/facebookresearch/esm/blob/main/environment.yml

meyersbs commented on var/spack/repos/builtin/packages/py-hatch-cython/package.py:
```diff
@@ -0,0 +1,23 @@
+
+    version("0.5.1", sha256="d01135e092544069c3e61f6dc36748ee369beacb893a5c43b9593a533f839703")
+
+    depends_on("python@3.8:", type=("build", "run"))
```
Missing `depends_on("py-typing-extensions", when="@:3.9", type=("build", "run"))`

meyersbs commented on var/spack/repos/builtin/packages/py-hatch-cython/package.py:
```diff
@@ -0,0 +1,23 @@
+    version("0.5.1", sha256="d01135e092544069c3e61f6dc36748ee369beacb893a5c43b9593a533f839703")
+
+    depends_on("python@3.8:", type=("build", "run"))
+
```
Missing dependency for py-hatch

meyersbs commented on var/spack/repos/builtin/packages/py-hatch-vcs/package.py:
```diff
@@ -13,10 +13,10 @@ class PyHatchVcs(PythonPackage):
     version("0.3.0", sha256="cec5107cfce482c67f8bc96f18bbc320c9aa0d068180e14ad317bbee5a153fee")
     version("0.2.0", sha256="9913d733b34eec9bb0345d0626ca32165a4ad2de15d1ce643c36d09ca908abff")
 
+    depends_on("py-hatchling@1.24.2:", when="@0.4:", type=("build", "run"))
```
Where did this version come from? I see `"hatchling>=1.1.0",`

ashim-mahara replied:
sorry, pebkam

meyersbs commented on var/spack/repos/builtin/packages/py-hatch/package.py:
```diff
@@ -0,0 +1,39 @@
+    depends_on("py-hyperlink@21.0.0:", type=("build", "run"))
+    depends_on("py-keyring@23.5.0:", type=("build", "run"))
+    depends_on("py-packaging@23.2:", type=("build", "run"))
+    depends_on("py-pexpect@4.8:", type=("build", "run"))
```
`"pexpect~=4.8",` --> `py-pexpect@4.8:4.8`

meyersbs commented on var/spack/repos/builtin/packages/py-hatch/package.py:
```diff
@@ -0,0 +1,39 @@
+
+    license("MIT")
+
+    version("1.12.0", sha256="ae80478d10312df2b44d659c93bc2ed4d33aecddce4b76378231bdf81c8bf6ad")
```
Missing dependency for `"rich>=11.2.0",`

meyersbs commented on var/spack/repos/builtin/packages/py-hatch/package.py:
```diff
@@ -0,0 +1,39 @@
+    depends_on("py-shellingham@1.4.0:", type=("build", "run"))
+    depends_on("py-tomli-w@1.0:", type=("build", "run"))
+    depends_on("py-tomlkit@0.11.1:", type=("build", "run"))
+    depends_on("py-userpath@1.7:", type=("build", "run"))
```
`"userpath~=1.7",` --> `py-userpath@1.7:1.7`

https://python-poetry.org/docs/dependency-specification/#version-constraints

meyersbs commented on var/spack/repos/builtin/packages/py-hatch/package.py:
```diff
@@ -0,0 +1,39 @@
+
+    depends_on("python@3.8:", type=("build", "run"))
+
+    depends_on("py-setuptools@42:", type="build")
```
Where did this dependency come from?

ashim-mahara replied:
pebkam, sorry.

meyersbs commented on var/spack/repos/builtin/packages/py-hatchling/package.py:
```diff
@@ -15,6 +15,8 @@ class PyHatchling(PythonPackage):
 
     license("MIT", checked_by="tgamblin")
 
+    version("1.25.0", sha256="7064631a512610b52250a4d3ff1bd81551d6d1431c4eb7b72e734df6c74f4262")
```
Check the dependencies for the two new versions of py-hatchling; they aren't the same as the versions below

meyersbs commented on var/spack/repos/builtin/packages/py-packaging/package.py:
```diff
@@ -13,7 +13,9 @@ class PyPackaging(PythonPackage):
 
     license("BSD-2-Clause")
-
+    version("24.1", sha256="026ed72c8ed3fcce5bf8950572258698927fd1dbda10a5e981cdf0ac37f4f002")
```
Missing `requires-python = ">=3.8"` for version 24.1

meyersbs commented on var/spack/repos/builtin/packages/py-uv/package.py:
```diff
@@ -0,0 +1,24 @@
+
+    depends_on("rust@1.81:", type=("build", "run"))
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-maturin@1.0:2.0", type=("build"))
```
Should be `@1.0:1`

meyersbs commented on var/spack/repos/builtin/packages/py-uv/package.py:
```diff
@@ -0,0 +1,24 @@
+
+    version("0.4.15", md5="bd21c6ca7c53f36fa514a9e7232acae0")
+
+    depends_on("rust@1.81:", type=("build", "run"))
```
Where did the version come from?

ashim-mahara replied:
https://github.com/astral-sh/uv/blob/0.4.15/Cargo.toml

meyersbs commented on var/spack/repos/builtin/packages/rust-bootstrap/package.py:
```diff
@@ -23,6 +23,17 @@ class RustBootstrap(Package):
     # should update these binary releases as bootstrapping requirements are
     # modified by new releases of Rust.
     rust_releases = {
+        "1.81.0": {
```
@alecbcs Can you review this?

alecbcs replied:
@ashim-mahara can you please break out `rust` and `rust-bootstrap` into their own PR?

meyersbs commented on var/spack/repos/builtin/packages/rust/package.py:
```diff
@@ -36,6 +36,7 @@ class Rust(Package):
     version("nightly")
 
     # Stable versions.
+    version("1.81.0", sha256="872448febdff32e50c3c90a7e15f9bb2db131d13c588fe9071b0ed88837ccfa7")
```
@alecbcs Can you review this?

ashim-mahara committed:
updated pacakges and their dependencies according to the PR review by @meyersbs

ashim-mahara committed:
typo fix for hatchling version; fix the minimum required setuptools version for evodiff

ashim-mahara committed:
added 1.9.0 and 1.7.0 userpath versions; required as a dependency

ashim-mahara committed:
added mlflow as a dependency

ashim-mahara committed:
changed biopython to an optional dependency according to review from @meyersbs; variant esmfold

ashim-mahara committed:
Updated Specs

- Pinned biotraj to 1:1 for py-biotite
- Added numpy and other dependencies for py-biotraj; they are dependent
  on the versions
- Excluded py-mlflow as a dependency for package py-evodiff; missing
  usage in
  package.
- Removed versioned dependencies from py-fair-esm
- Added a version to py-packaging
- Added py-setuptools as a dependency in py-userpath
- Added sha256 as hashes for py-uv

meyersbs reviewed (changes requested):
Please post the comment `@spackbot fix style` so you pass style checks.

I will test the build soon.

``` mermaid
graph TD;
  'py-evodiff'-->'py-sequence-models';
  'py-evodiff'-->'py-blosum';
  'py-evodiff'-->'py-fair-esm';
  'py-evodiff'-->'py-biotite';
  'py-evodiff'-->'py-pdb-tools';
  'py-biotite'-->'py-hatch';
  'py-biotite'-->'py-hatchling';
  'py-biotite'-->'py-hatch-vcs';
  'py-biotite'-->'py-hatch-cython';
  'py-biotite'-->'py-biotraj';
  'py-hatch'-->'py-hatchling';
  'py-hatch'-->'py-hatch-vcs';
  'py-hatch'-->'py-packaging';
  'py-hatch'-->'py-userpath';
  'py-hatch'-->'py-uv';
  'py-hatch'-->'py-virtualenv';
  'py-hatchling'-->'py-packaging';
  'py-hatch-vcs'-->'py-hatchling';
  'py-hatch-cython'-->'py-hatchling';
  'py-hatch-cython'-->'py-hatch';
  'py-userpath'-->'py-hatchling';
```

meyersbs commented:
Depends on https://github.com/spack/spack/pull/46597 (merged)

ashim-mahara commented:
@spackbot fix style

ashim-mahara committed:
style changes

meyersbs reviewed (approved):
Builds, installs, and imports for me. I'm comfortable approving this.

```
==> Installing py-evodiff-1.1.0-gjvtgluzghjprzjzg3jk7kf2klckc2sy [233/233]
==> No binary for py-evodiff-1.1.0-gjvtgluzghjprzjzg3jk7kf2klckc2sy found: installing from source
==> Fetching https://files.pythonhosted.org/packages/source/e/evodiff/evodiff-1.1.0.tar.gz
==> No patches needed for py-evodiff
==> py-evodiff: Executing phase: 'install'
==> py-evodiff: Successfully installed py-evodiff-1.1.0-gjvtgluzghjprzjzg3jk7kf2klckc2sy
  Stage: 0.50s.  Install: 2.22s.  Post-install: 0.18s.  Total: 3.54s
[+] /dev/shm/bsmits_spack/opt/spack/linux-rhel9-skylake_avx512/gcc-13.2.0/py-evodiff-1.1.0-gjvtgluzghjprzjzg3jk7kf2klckc2sy

$ spack load py-evodiff
$ python3
Python 3.11.9 (main, Oct  3 2024, 09:02:54) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import evodiff
>>> import esm
>>> import blosum
>>> import biotite
>>> import sequence_models
>>> import pdb
>>> import biotraj
```

tgamblin commented:
Closing and reopening to fix pipeline issues.

meyersbs commented:
@tgamblin The CI is failing for this, but I don't think it's related to the contents of the PR. Can you take a look and let @ashim-mahara know if there's anything he needs to change?

alecbcs reviewed (approved):
Looks good to me. Thanks @ashim-mahara!
